### PR TITLE
⬆️ Upgrade to FHIR version R4

### LINKED ISSUE #59: Upgrade to FHIR R4
I'm starting a new issue for this just so we can keep track of the rational for our decision. 

When I first started looking into FHIR there were a couple reasons I opted to start with STU3 rather than R4:

1. Forge R4 was not able to connect to simplifier and load profiles (e.g. US Core, Simplifier Core, etc). It looks like they just released (yesterday!) a version of Forge that is able to do this: https://simplifier.net/organization/firely/news/76

2. Vonk R4 - I didn't see a way to access an R4 docker image license for Vonk. Currently, on my STU3 Simplifier project I can go to my project home page > Download > NatashaSTU3Workspace FHIR Server and download the zip file containing the Vonk docker compose file and license for FHIR STU3. This is not available for a R4 Simplifier project (I don't seem to have that same button on my R4 project page). 

I also tried to go to the simplifier Docker Hub page to look for R4 images, but there isn't a clear way to tell whether the image tags correspond to FHIR version numbers.

There is some talk of Vonk R4 on Zulip fhir chat, so it might exist. I'm going to post a question to see if a Vonk R4 docker image exists.

### PR DESCRIPTION
**Pull request by znatty22**
Closes #59 

I won't merge this until we all agree to upgrade to R4

**TODO - Before merging**
1. Replace Kids First STU3 simplifier project with a Kids First R4 simplifier project (or pay for Simplifier so we can have more than 1 project in the account)

**STU3 Profile/Resource Migration to R4**
1. All StructureDefinition resources must have the `fhirVersion=4.0.0` key, value pair.
2. No change needed for non-StructureDefinition resources

**Notes**
If you try to send any request (GET, POST, PUT, PATCH, etc) without the correct Content-Type (application/fhir+json; fhirVersion=4.0) header to the R4 Vonk server, it throws an exception. This is probably a bug and needs to be reported. Not sure how to do that yet. 

The CLI adds this header to every request so this isn't an issue when using the CLI to do validation. However, you should keep this in mind if you use something else (e.g. Postman) to exchange data w the R4 Vonk server.

### CODE REVIEW

**Comment by allisonheath** [comment]
👍 to upgrade to R4 - thanks for listing the current caveats.

**Comment by fiendish** [comment]
> *If you try to send any request (GET, POST, PUT, PATCH, etc) without the correct Content-Type (application/fhir+json; fhirVersion=4.0) header to the R4 Vonk server, it throws an exception. This is probably a bug*

The spec looks kinda funny to me on that point. It says clients SHALL send the correct header but then indicates what the server SHOULD do if they don't.

**Comment by znatty22** [comment]
> The spec looks kinda funny to me on that point. It says clients SHALL send the correct header but then indicates what the server SHOULD do if they don't.

I think regardless, the server shouldn't have an unhandled exception. That's what I meant by I think its a bug. Its interesting bc if you look in the release notes for the Vonk beta2 release, it sounds like they fixed the issue or maybe something similar.

File: README.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ contains FHIR profiles, example resources, and implementation guides for
 a specific version of FHIR. See https://simplifier.net/learn.
 
 Everything we put in this folder will eventually be published to the corresponding
-[KidsFirstSTU3 Simplifier Project](https://simplifier.net/kidsfirststu3) so that
+[KidsFirstR4 Simplifier Project](https://simplifier.net/kidsfirstr4) so that
```
Comment (ShahimEssaid) [inline README.md]: The case difference between the URL project name and the SIMPLIFIER_PROJECT_NAME variable probably doesn't matter.

Comment (znatty22) [reply README.md]: Correct, case doesn't matter

**Pull request merged**